modules/data-catalog-tag/main.tf

```diff
@@ -31,8 +31,13 @@ resource "google_data_catalog_tag" "engine" {
   dynamic "fields" {
     for_each = each.value.fields
     content {
-      field_name   = fields.key
-      string_value = fields.value
+      field_name      = fields.key
+      display_name    = try(fields.value.display_name, null)
+      order           = try(fields.value.order, null)
+      string_value    = try(fields.value.string_value, null)
+      double_value    = try(fields.value.double_value, null)
+      enum_value      = try(fields.value.enum_value, null)
+      timestamp_value = try(fields.value.timestamp_value, null)
```

Comment:
As `fields.value` is never null, can we just skip `try(...)` here? All other fields just just resolve to `null`. That would allow to capture, that `order` and `display_name` is not defined in the variable :-)

Reply:
Order should not be there, it was a mistake. We need to keep try() because in the yaml definition the user may put just one of them.

Reply:
Ah understood, the approach to avoid `try(...)` in this situation is to go through data from yaml and merge each element with map which has all of the objects with null value